src/vulkan/resource.h

```diff
@@ -68,14 +69,7 @@ class Resource {
 
   Result MapMemory(VkDeviceMemory memory);
   void UnMapMemory(VkDeviceMemory memory);
-
-  // Set |memory_ptr_| as |ptr|. This must be used for only push constant.
-  // For Vulkan buffer and image i.e., Buffer and Image classes, we should
-  // not call this but uses MapMemory() method.
   void SetMemoryPtr(void* ptr) { memory_ptr_ = ptr; }
-
```

Comment:
This one needs a comment:

>  Records a memory barrier on |command|, to ensure prior writes to this buffer have completed and are available and to subsequent commands.


IMHO this is a critical thing to get right: it async work to be done later.
In fact the parameter name is bad, it should be |command_buffer| instead of |command|.

Reply:
done